File: crypto-casino-backend/src/main/java/hu/bme/aut/crypto_casino_backend/dto/game/SpinResponse.java
package hu.bme.aut.crypto_casino_backend.dto.game;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SpinResponse {
    private Long spinId;
    private int[] reels;
    private BigDecimal betAmount;
    private BigDecimal winAmount;
    private boolean isWin;
    private LocalDateTime timestamp;
    private BigDecimal newBalance;
}

File: crypto_casino_android/app/src/main/java/hu/bme/aut/crypto_casino_android/data/model/slot/SpinResponse.kt
package hu.bme.aut.crypto_casino_android.data.model.slot

import java.math.BigDecimal
import java.time.LocalDateTime

data class SpinResponse(
    val spinId: Long,
    val reels: List<Int>,
    val betAmount: BigDecimal,
    val winAmount: BigDecimal,
    val timestamp: LocalDateTime
)